### ISSUE TITLE
Mardown preview: Checkbox lists with sub-items without checkbox not rendered correctly

### ISSUE THREAD

**Issue by phistep** [zed-industries/zed#19944]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

1. Create a new document and save it with `.md` extension
2. Create a list with a to-do item, then a child list without
   ```markdown
   - [ ] to-do item
     - sub-item without checkbox
     - another
   ```
3. run ` markdown: open preview` from the command palette

Expected:

> The `to-do item` has a checkbox, the `sub-item without checkbox` and `another` items do not

Actual:

> The `to-do item` does not have a checkbox, the `sub-item withouth checkbox` does and `another` does not

![image](https://github.com/user-attachments/assets/f640bcc3-ec53-4270-8ad6-3e1c44820d43)

### Environment

Zed: v0.158.1 (Zed)
OS: Linux Wayland fedora 40
Memory: 30.8 GiB
Architecture: x86_64
GPU: Intel(R) Arc(tm) Graphics (MTL) || Intel open-source Mesa driver || Mesa 24.1.7

**Comment by bennetbo** [comment]
Thanks for reporting, this was fixed in #19383 and is in Zed Preview right now, landing on stable later today

**Comment by phistep** [comment]
Sorry, next time I will check the preview channel first!
